ast/annotations_test.go

```diff
@@ -1095,6 +1097,319 @@ func TestAnnotations_toObject(t *testing.T) {
 	}
 }
 
+func TestNewAnnotationsSet_WithOptions(t *testing.T) {
```

Comment:
I'm not sure about this test name, how about `TestNewAnnotationsRef_JSONOptions`?

Also, I think that this test might be best located in `ast/marshal_test.go`. We added some pretty similar tests in there and I think it'd make sense for this to be located with them.